src/tespy/tools/analyses.py

```diff
@@ -0,0 +1,759 @@
+
+        Method is recursively called if a group was removed from display to
+        catch cases, where multiple groups are attached in line without
+        any streams leaving the line.
+
+        Parameters
+        ----------
+        group_data : dict
+            Dictionary containing the modified component group data.
+        """
+        for fkt_group, data in group_data.copy().items():
+            source_group = self.single_group_input(fkt_group, group_data)
+            if source_group is not None and len(group_data[fkt_group]) == 1:
+                target_group = group_data[fkt_group].index[0]
+                target_value = group_data[fkt_group].loc[target_group]
+                group_data[source_group].loc[target_group] = target_value
+                group_data[source_group].loc[target_group] = target_value
+                group_data[source_group].drop(fkt_group, inplace=True)
+                del group_data[fkt_group]
+                # recursive call in case multiple components are attached in
+                # a line without any conversion
+                self.remove_transit_groups(group_data)
+
+    def generate_plotly_sankey_input(
+            self, node_order=[], colors={}, display_thresold=1e-3):
+        """Generate input data for sankey plots.
+
+        Only exergy flow above the display threshold is included. All
+        component groups with transit only (one input and one output) are cut
+        out of the display.
+
+        Parameters
+        ----------
+        node_order : list
+            Order for the nodes in the sankey diagram (optional).
+            In case no order is passed, a generic order will be used.
+
+        colors : dict
+            Dictionary containing a color for every stream type (optional).
+            Stream type is the key, the color is the corresponding value.
+            Stream types are:
+
+            - :code:`E_F`, :code:`E_P`, :code:`E_L`, :code:`E_D`
+            - names of the pure fluids of the tespy Network
+            - :code:`mix` (used in case of any gas mixture)
+            - labels of internal busses
+
+            In case no colors are passed, the matplotlib :code:`Set1` colormap
+            will be applied.
+
+        Returns
+        -------
+        tuple
+            Tuple containing the links and node_order for the plotly sankey
+            diagram.
+        """
+        group_data = self.group_data.copy()
+
+        for fkt_group, data in self.group_data.items():
+            group_data[fkt_group] = group_data[fkt_group][
+                data['value'].abs() >= display_thresold]
+
+        self.remove_transit_groups(group_data)
+
+        if len(node_order) == 0:
+            node_order = (
+                ['E_F'] + [b.label for b in self.E_F] +
+                [fkt_group for fkt_group in self.group_overview.index] +
+                [b.label for b in self.internal_busses + self.E_P + self.E_L] +
+                ['E_P', 'E_L', 'E_D'])
```

Comment:
Node order should be checked for exact length. If length is larger than zero but smaller than total amount of nodes, an error should be thrown.